src/service/getIplayerExecutableService.ts

```diff
@@ -151,11 +155,24 @@ export class GetIplayerExecutableService {
         for (const line of lines) {
             if (line.startsWith('RESULT|:|')) {
                 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-                const [_, pid, rawTitle, seriesStr, episodeStr, number, channel, durationStr, onlineFrom] = line.split('|:|');
-                const episode: number | undefined = (episodeStr == '' ? undefined : parseInt(episodeStr));
-                const [title, series] = (seriesStr == '' ? [rawTitle, undefined] : extractSeriesNumber(rawTitle, seriesStr))
-                const type: VideoType = episode && series ? VideoType.TV : VideoType.MOVIE;
-                const size: number | undefined = durationStr ? parseInt(durationStr) * sizeFactor : undefined;
+                const [_, pid, rawTitle, seriesStr, episodeStr, number, channel, durationStr, onlineFrom, epTitle] = line.split('|:|');
+                const [ title, episodeNum, seriesNum ] = parseEpisodeDetailStrings(rawTitle, episodeStr, seriesStr)
+                const [ type, episode, episodeTitle, series ] = calculateSeasonAndEpisode({
```

Comment:
Pass what we have to the function the native search uses, to reuse the same logic. Not entirely happy with this, may refactor it later.